Pod/Core/TypeConversion.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  TypeConversion.swift
+//  Pods
+//
+//  Created by Sam Dean on 4/4/17.
+//
+//
+
+import Foundation
+
+
+public protocol FromStringable {
```

Comment:
Right, I think I've got this now - check out 85bf316.

I've made the protocol `MatchedStringRepresentable`, with an init method with a parameter name, so it's unambiguous and won't clash with other methods.

I've extended it, but only for types which also implement `LosslessStringConvertible` so `Int` and `String` can implicitly implement `MatchedStringRepresentable`.

I've overriden the `init?(matchedString:)` method in `Bool`, even though it has the one from the extension, so that it's case insensitive.

I've implemented the `init?(matchedString:)` method in `Int`, calling through to `init(_:radix:)` with a radix of 10.

Let me know what you think.